docker/build.gradle

```diff
@@ -44,7 +56,23 @@ task quickstartNuke(type: Exec, dependsOn: ":metadata-ingestion:install") {
     commandLine 'bash', '-c',  cmd.join(" ")
 }
 
-task quickstartRebuild() {
-    dependsOn(quickstart_modules.collect { it + ':clean' })
-    dependsOn'quickstart'
+task quickstartDebug(type: Exec, dependsOn: ':metadata-ingestion:install') {
+    dependsOn(debug_modules.collect { it + ':dockerTagDebug' })
+    shouldRunAfter ':metadata-ingestion:clean', 'quickstartNuke'
+
+    environment "DATAHUB_TELEMETRY_ENABLED", "false"
+    environment "DOCKER_COMPOSE_BASE", "file://${rootProject.projectDir}"
+
+    def cmd = [
+            'source ../metadata-ingestion/venv/bin/activate && ',
+            'datahub docker quickstart',
+            '--version', "debug",
+            '--dump-logs-on-failure'
+    ] + debug_compose_args
+    commandLine 'bash', '-c', cmd.join(" ")
+}
+
+task debugReload(type: Exec) {
```

Comment:
One minor naming thing - I'd assume that this deployment mode would also set debug mode in the GMS / Frontend containers so I could attach a debugger to it. But I don't think it does. Actually we support this via a values.yaml Helm parameter which in turn sets a JAVA_OPTS environment variable containing the debugger configuration... Just something to think about...

Reply:
Definitely thought about this, I'd like to add this as a flag because it definitely slows performance.